File: src/main/java/com/hubermjonathan/discord/common/models/BotOwnerButton.java
package com.hubermjonathan.discord.common.models;

public abstract class BotOwnerButton extends Button {
    public BotOwnerButton(String id) {
        super(id);
    }

    @Override
    protected boolean shouldIgnoreEvent() {
        return getEvent().getUser().isBot()
                || !getEvent().getComponentId().equals(getId())
                || !getEvent().getUser().getId().equals(System.getenv("BOT_OWNER_ID"));
    }
}

File: src/main/kotlin/com/hubermjonathan/discord/common/models/BotOwnerButton.kt
package com.hubermjonathan.discord.common.models

import com.hubermjonathan.discord.mitch.botOwner
import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent

abstract class BotOwnerButton(id: String, featureContext: FeatureContext) : Button(id, featureContext) {
    override fun shouldIgnoreEvent(event: ButtonInteractionEvent): Boolean {
        val userIsNotBotOwner = event.user != jda.botOwner

        return super.shouldIgnoreEvent(event) ||
            userIsNotBotOwner
    }
}
